### PR TITLE
Get closest color if TrueColor not supported

### PR DESCRIPTION
**Pull request by spenserblack**
This PR makes TrueColor convert into one of the more "basic" colors (`Color::Black`, `Color::Red`, etc.) if TrueColor is not supported based on the `COLORTERM` environment variable. It gets the closest color based on Euclidean distance.

I'm not sure if some of the functions I added should be made public or not, so I left them private for now.

### CODE REVIEW

**Review by kurtlawrence — CHANGES_REQUESTED** [review]
Please add some unit tests to confirm behaviour.

File: src/color.rs [hunk 1]
```diff
@@ -212,4 +292,37 @@ mod tests {
...
+
+        make_euclidean_distance_test! { almost_black: (10, 15, 10), Color::Black }
+        make_euclidean_distance_test! { almost_red: (215, 10, 10), Color::Red }
+        make_euclidean_distance_test! { almost_green: (10, 195, 10), Color::Green }
+        make_euclidean_distance_test! { almost_yellow: (195, 215, 10), Color::Yellow }
+        make_euclidean_distance_test! { almost_blue: (0, 0, 200), Color::Blue }
+        make_euclidean_distance_test! { almost_magenta: (215, 0, 195), Color::Magenta }
+        make_euclidean_distance_test! { almost_cyan: (10, 215, 215), Color::Cyan }
+        make_euclidean_distance_test! { almost_white: (209, 209, 229), Color::White }
+    }
```
Comment (spenserblack) [inline src/color.rs:327]: >Please add some unit tests to confirm behaviour.

No problem, but could you tell me what these tests are lacking?

Comment (kurtlawrence) [reply src/color.rs:327]: Yeah for the public facing API (`to_[fg,bg]_str`), just to avoid having a future change break it somehow.

Thanks for the PR!

Comment (kurtlawrence) [reply src/color.rs:327]: Hmmm, nah I don't like the idea of that, at least not for this PR.

The code is good, I was just trying to protect from unintentional changes in behaviour from future updates.

**Review by kurtlawrence — APPROVED** [review]

**Pull request merged**